examples/plot_partial_dependence_visualization_api.py

```diff
@@ -0,0 +1,123 @@
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.pipeline import make_pipeline
+from sklearn.tree import DecisionTreeRegressor
+from sklearn.inspection import plot_partial_dependence
+
+
+##############################################################################
+# Train models on the boston housing price dataset
+# ================================================
+#
+# First, we load the boston housing price dataset and split the the dataset
+# into a training and test set. Then, we train a decision tree and a
+# multi-layer perceptron on the training set.
+
+boston = load_boston()
+X, y = boston.data, boston.target
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1,
+                                                    random_state=0)
+
+tree = DecisionTreeRegressor()
+mlp = make_pipeline(StandardScaler(),
+                    MLPRegressor(hidden_layer_sizes=(100, 100),
+                                 tol=1e-2, max_iter=500, random_state=0))
+tree.fit(X_train, y_train)
+mlp.fit(X_train, y_train)
+
+##############################################################################
+# Plotting partial dependence of the two models independently
+# ===========================================================
+#
+# Next, we plot a partial dependence curves for features "LSTAT" and "RM" for
+# for the decision tree.
+fig, ax = plt.subplots(figsize=(12, 6))
+ax.set_title("Decision Tree")
+tree_disp = plot_partial_dependence(tree, X_test, ["LSTAT", "RM"],
+                                    feature_names=boston.feature_names, ax=ax)
+
+##############################################################################
+# The partial depdendence curves can be plotted for the multi-layer perceptron.
+# In this case `line_kw` was passed to
+# `~sklearn.inspection.plot_partial_dependence` to change the color of the
+# curve and `n_cols` was set to 1 to set the number of columns to 1.
+fig, ax = plt.subplots(figsize=(8, 8))
+ax.set_title("Multi-layer Perceptron")
+mlp_disp = plot_partial_dependence(mlp, X_test, ["LSTAT", "RM"],
+                                   feature_names=boston.feature_names, ax=ax,
+                                   n_cols=1, line_kw={"c": "red"})
+
+##############################################################################
+# Plotting partial dependence of the two models together
+# ======================================================
+#
+# The `tree_disp` and `mlp_disp`
+# :class:`~sklearn.inspection.PartialDependenceDisplay` objects contain all the
+# computed information needed to recreate the partial dependence curves. This
+# means we can easily create additional plots without needing to recompute the
+# curves.
+#
+# One way to plot the curves is to place them in the same figure, with the
+# curves of each model on each row. First, we create a figure with two axes
+# within two rows and one column. The two axes are passed to the
```

Comment:
I like the example but it's pretty condensed and it combines multiple functionalities at once so it's hard to understand how things interact independently.

For example it's a bit confusing at first how you can end up with a 2 by 2 grid while you only asked for 1 column.

I think it's missing a very simple plot with just 1 feature where you combine the curves of the 2 models. Then

Reply:
I added a "Plotting partial dependence for one feature" section at the end. With the current API, there are two ways to plot a single feature:

### Option 1
```py
tree_disp = plot_partial_dependence(tree, X, ["LSTAT"],
                                    feature_names=feature_names)
mlp_disp = plot_partial_dependence(mlp, X, ["LSTAT"],
                                   feature_names=feature_names,
                                   ax=tree_disp.axes_, line_kw={"c": "red"})
```

### Option 2
```py
_, ax = plt.subplots()
tree_disp = plot_partial_dependence(tree, X, ["LSTAT"],
                                    feature_names=feature_names, ax=[ax])
mlp_disp = plot_partial_dependence(mlp, X, ["LSTAT"],
                                   feature_names=feature_names,
                                   ax=[ax], line_kw={"c": "red"})
```

For the example, I went with option 1. The "nicer" way to do this is:

### Possible option that is being disallowed.
```py
_, ax = plt.subplots()
tree_disp = plot_partial_dependence(tree, X, ["LSTAT"],
                                    feature_names=feature_names, ax=ax)
mlp_disp = plot_partial_dependence(mlp, X, ["LSTAT"],
                                   feature_names=feature_names,
                                   ax=ax, line_kw={"c": "red"})
```

The first call will call `ax.set_visible(False)` to denote that the space has been used. The second call will see that the axes is not visible and raise an error. We can technically support this "single feature" and "single axes" case, which I think will add another layer of complexity to the API i.e. "If `ax` is a single axes and `len(features) == 1`, then we behave differently"